strongbox-storage/strongbox-storage-layout-providers/strongbox-storage-maven-layout-provider/src/test/java/org/carlspring/strongbox/io/ArtifactOutputStreamTest.java

```diff
@@ -42,13 +42,22 @@
 {
 
     public static final String REPOSITORY_RELEASES = "aos-releases";
+    
+    private static boolean initialized;
```

Comment:
@sbespalov I found couple of issues and the problem is not solved completely yet. Here are the issues/tips:
* Stop using `@PreDestroy` inside Spring test classes. It is useless and not even invoked (see https://github.com/strongbox/strongbox/wiki/Writing-Tests#spring-test-tips)
* Many resources (readers/inputStreams) are not closed in the source code - thus some processes holds files unnecessarily
* `org.carlspring.strongbox.testing.TestCaseWithRepository#removeRepositories `does not remove the repositories physically, so they shouldn't be created once again in the `org.carlspring.strongbox.services.MavenChecksumServiceTest#initialize` method again (regardless `initialized `flag). 

cc: @carlspring 

Here is partial PR for those issues: https://github.com/sbespalov/strongbox/pull/1/files. Now the exception is:

> java.io.IOException: Unable to delete directory C:\Users\fuss\IdeaProjects\strongbox\strongbox-storage\strongbox-storage-layout-providers\strongbox-storage-maven-layout-provider\target\strongbox-vault\storages\storage0\mcs-releases\org\carlspring\strongbox\checksum\maven\strongbox-checksum.

I bet there is some process which keeps the directory open. It should be fixed tomorrow.